controller/resource.go

```diff
@@ -126,18 +154,23 @@ func (s ServiceStateLister) fetchECSTasks() (map[string]struct{}, error) {
 			}
 
 			if !isMeshTask(task) {
-				s.Log.Info("skipping non-mesh task", "task-arn", task.TaskArn)
+				s.Log.Info("skipping non-mesh task", "task-arn", *task.TaskArn)
 				continue
 			}
 
-			serviceName, err := serviceNameForTask(task)
+			serviceName, err := s.serviceNameForTask(task)
 
 			if err != nil {
 				s.Log.Error("couldn't get service name from task", "task-arn", task.TaskArn, "tags", task.Tags, "err", err)
 				continue
 			}
 
-			resources[serviceName] = struct{}{}
+			if serviceName.Partition() != s.Partition {
+				s.Log.Info("skipping task in external partition", serviceName.Partition(), "task-arn", *task.TaskArn)
+				continue
+			}
+
+			resources[serviceName.String()] = serviceName
```

Comment:
That would be awesome. Just to clarify, we should make our own struct that is similar to `ServiceName` rather than import and depend on it. As far as I know, that part of Consul isn't supposed to be used as a library

Reply:
Moved to new `ServiceName` struct and now use it for the key in the mappings.